### PR TITLE
Update templates to make use of `getModelSchemaRef`

### PR DESCRIPTION
**Pull request by nabdelgadir**
- Update CLI templates to use `getModelSchemaRef`
- Update examples and docs to use it too 

Closes https://github.com/strongloop/loopback-next/issues/3204.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/cli/generators/relation/templates/controller-relation-template-belongs-to.ts.ejs [hunk 1]
```diff
@@ -4,6 +4,7 @@ import {
 import {
   param,
   get,
+  getModelSchemaRef
```
Comment (raymondfeng) [inline packages/cli/generators/relation/templates/controller-relation-template-belongs-to.ts.ejs]: It should be `getModelSchemaRef,`

File: packages/cli/generators/relation/templates/controller-relation-template-belongs-to.ts.ejs [hunk 2]
```diff
@@ -23,7 +24,7 @@ export class <%= controllerClassName %> {
         description: '<%= targetModelClassName %> belonging to <%= sourceModelClassName %>',
         content: {
           'application/json': {
-            schema: { type: 'array', items: { 'x-ts-type': <%= targetModelClassName %> } },
+            schema: { type: 'array', items: getModelSchemaRef(<%= targetModelClassName %>) },
```
Comment (bajtos) [inline packages/cli/generators/relation/templates/controller-relation-template-belongs-to.ts.ejs]: While we are touching this file & line, should we fix the white-space formatting too?

```suggestion
            schema: {type: 'array', items: getModelSchemaRef(<%= targetModelClassName %>)},
```

File: packages/cli/generators/relation/templates/controller-relation-template-has-many.ts.ejs [hunk 3]
```diff
@@ -32,7 +32,7 @@ export class <%= controllerClassName %> {
         description: 'Array of <%= targetModelClassName %>\'s belonging to <%= sourceModelClassName %>',
         content: {
           'application/json': {
-            schema: { type: 'array', items: { 'x-ts-type': <%= targetModelClassName %> } },
+            schema: { type: 'array', items: getModelSchemaRef(<%= targetModelClassName %>) },
```
Comment (bajtos) [inline packages/cli/generators/relation/templates/controller-relation-template-has-many.ts.ejs]: Similarly here - should we fix the white-space formatting too?

```suggestion
            schema: {type: 'array', items: getModelSchemaRef(<%= targetModelClassName %>)},
```

File: packages/cli/generators/relation/templates/controller-relation-template-has-many.ts.ejs [hunk 4]
```diff
@@ -49,7 +49,7 @@ export class <%= controllerClassName %> {
     responses: {
       '200': {
         description: '<%= sourceModelClassName %> model instance',
-        content: { 'application/json': { schema: { 'x-ts-type': <%= targetModelClassName %> } } },
+        content: { 'application/json': { schema: getModelSchemaRef(<%= targetModelClassName %>) } },
```
Comment (bajtos) [inline packages/cli/generators/relation/templates/controller-relation-template-has-many.ts.ejs]: Similarly here - should we fix the white-space formatting too?

```suggestion
        content: {'application/json': {schema: getModelSchemaRef(<%= targetModelClassName %>)}},
```

**Pull request merged**